controllers/configurationpolicy_controller.go

```diff
@@ -1598,6 +1600,24 @@ func (r *ConfigurationPolicyReconciler) handleObjects(
 			} else {
 				resultEvent.compliant = false
 				resultEvent.reason = reasonWantFoundDNE
+				// Length of objNames = 0, complianceType == musthave and the name field is empty
+				// Find Noncompliant resouces to add to the status.relatedObjects
+				if objDetails.kind != "" && objDetails.name == "" {
+					objNames = getNamesOfKind(
+						nil,
+						mapping.Resource,
+						objDetails.isNamespaced,
+						namespace,
+						r.TargetK8sDynamicClient,
+						strings.ToLower(string(objectT.ComplianceType)),
+						true,
+					)
+
+					// Change reason to Resource found but does not match
+					if len(objNames) > 0 {
+						shouldAddCondensedRelatedObj = true
```

Comment:
Why not just always set this to true instead of duplicating the list query performed in `getNamesOfKind`?

Reply:
len(objNames) == 0 there is nothing showing on relatedobj.  in other words there is nothing to debug

Reply:
I'd prefer if whenever the object template is not named and it's noncompliant, you just set `shouldAddCondensedRelatedObj = true`. Perhaps there are no objects, but that's okay. It makes the behavior consistent.

Also, I'm not sure if `reasonWantFoundNoMatch` is better than `reasonWantFoundDNE`, so if that's why you are duplicating the list query by calling `getNamesOfKind` again, then it's not worth it in my opinion.

Reply:
um... to me, `reasonWantFoundNoMatch` is better. Because the name is * in relatedObj and users only see the `reason` on the console. `reasonWantFoundNoMatch` is for relatedObj.. isn't it? @JustinKuli  how about yoiu

Reply:
Sorry for slow response, if you're still looking for input, I think `reasonWantFoundNoMatch` makes more sense when the name is `*`. It doesn't make as much sense to say something "resource *, not found" when some instances of that resource do exist.

So for `mustnothave`, same kind of thing: if the names would be `*` or something similar, because some resources were found they just didn't fully match the policy criteria, then use a `NoMatch` kind of reason.